edit-post/api/more-menu-item.js

```diff
@@ -0,0 +1,94 @@
+/* eslint no-console: [ 'error', { allow: [ 'error' ] } ] */
+
+/**
+ * External dependencies
+ */
+
+/**
+ * Internal dependencies
+ */
+import { applyFilters } from '@wordpress/hooks';
+import { isString } from 'util';
+import { validatePluginId } from './utils';
+import { activateSidebar } from './sidebar';
+
+const menuItems = {};
+
+/**
+ * Registers a plugin under the more menu.
+ *
+ * @param {string}   menuItemId        The unique identifier of the plugin. Should be in
+ *                                     `[namespace]/[name]` format.
+ * @param {Object}   settings          The settings for this menu item.
+ * @param {string}   settings.title    The name to show in the settings menu.
+ * @param {func}     settings.target   The registered plugin that should be activated.
+ * @param {string}   [settings.icon]   SVG Icon url.
+ *
+ * @return {Object} The final sidebar settings object.
+ */
+export function registerMoreMenuItem( menuItemId, settings ) {
```

Comment:
Should it be renamed from `menuItemId` to `pluginId` given that it is validated as it were the plugin id?

Reply:
Depends on whether you want to view a menu item as a plugin, although I do think it's a good idea to validate any ID provided by plugin developers in the same way.